libminifi/test/TestBase.h

```diff
@@ -445,6 +445,13 @@ class TestController {
     return dir;
   }
 
+  template<size_t N>
+  utils::Path createTempDirectory(const char (&format)[N]) {
```

Comment:
This simplifies things a bit :+1:

Reply:
if only there wasn't now three different ways to create a temporary directory (now 2 in TestController, and 1 standalone in TestUtils.h) but I felt like removing all the single-use `format` variables in all the tests deserves its own PR

Reply:
Yes I thought the same, it should definitely be in a separate PR.

Reply:
https://issues.apache.org/jira/browse/MINIFICPP-1600